### PR TITLE
cli: Add versions of all pods in kadalu-namespace

### LINKED ISSUE #675: Add debug stats/logs for all kadalu pods to ask user for easy reference
Since Kadalu already has logs option making users easier and faster for debug. It would be better to include kadalu version info as well to it.
Quote by @amarts:
"_@aravindavk @leelavg can we get commands common for diagnosing into kubectl kadalu debug or something similar, so instead of asking all these question, we ask users to share output of this one command, which would capture logs, get version info, and more so its easier ?_ "

### PR DESCRIPTION
**Pull request by vatsa287**
This PR adds versions of all pods managed by kadalu,
by exec into them. Helps while debugging for version
related errors.

Fixes: #675 
Signed-off-by: Shree Vatsa N <vatsa@kadalu.io>

### CODE REVIEW

**Comment by vatsa287** [comment]
TEST
```
vatsa@thinkpad:~/kadalu/cli/kubectl_kadalu$ python3 __main__.py logs | grep "kadalu version" -i -B2
----- (Kadalu Namespace) pod/kadalu-csi-nodeplugin-d5j6n -----
----- (kadalu Version) devel -----

----- (Kadalu Namespace) pod/kadalu-csi-provisioner-0 -----
----- (kadalu Version) devel -----

----- (Kadalu Namespace) pod/operator-7dfcc8fb86-xbtvm -----
----- (kadalu Version) devel -----

----- (Kadalu Namespace) pod/server-storage-pool-1-0-0 -----
----- (kadalu Version) devel -----
```

**Comment by leelavg** [comment]
1. This doesn't entirely fix the referenced issue, can you pls clone that comment and create a new issue?
2. All this time we were asking users to provide logs but missed the detail the kadalu cli has already logs option 
3. Maybe we can add cli binary to operator itself and create necessary subcommands @amarts ?

**Comment by leelavg** [comment]
- I think `version` command is a good place for current implementation 
- Like `k kadalu version` displays:
```
Plugin: 0.8.6
Operator: 0.8.6
CSI controller: ...
CSI provisioner: ...
Server: 0.8.6
```
- How'd that be?

**Comment by vatsa287** [comment]
@leelavg 
```
vatsa@thinkpad:~/kadalu/cli/kubectl_kadalu$ python3 __main__.py version
kubectl-kadalu plugin: devel
kadalu pod(s) versions
pod/kadalu-csi-nodeplugin-w2nxt: devel
pod/kadalu-csi-provisioner-0: devel
pod/operator-7dfcc8fb86-dr72n: devel
pod/server-storage-pool-1-0-0: devel
```

File: cli/kubectl_kadalu/__main__.py [hunk 1]
```diff
@@ -39,9 +39,54 @@ def get_args():
...
+    except utils.CommandError as err:
+        utils.command_error(cmd, err.stderr)
+        return None
+
+
+def get_kadalu_version_in_pod(pod):
+    """ Get kadalu version from pods inside kadalu-namespace by exec """
+
+    try:
+        cmd = ["kubectl", "exec", "-nkadalu", pod, "--", "python",
```
Comment (leelavg) [inline cli/kubectl_kadalu/__main__.py]: A simple `echo $KADALU_VERSION` also should work, can you pls check?

**Review by leelavg — COMMENTED** [review]
There's a room for extra enhancements but for now this functionality suffice

**Comment by aravindavk** [comment]
> @leelavg
> 
> ```
> vatsa@thinkpad:~/kadalu/cli/kubectl_kadalu$ python3 __main__.py version
> kubectl-kadalu plugin: devel
> kadalu pod(s) versions
> pod/kadalu-csi-nodeplugin-w2nxt: devel
> pod/kadalu-csi-provisioner-0: devel
> pod/operator-7dfcc8fb86-dr72n: devel
> pod/server-storage-pool-1-0-0: devel
> ```

Output looks good. Make sure the version command does not fail if no Kadalu Pods are installed or Kubernetes cluster is not running.

File: cli/kubectl_kadalu/__main__.py (hunk 1 above)
Comment (vatsa287) [reply cli/kubectl_kadalu/__main__.py]: Yes it works.
```
vatsa@thinkpad:~/kadalu/cli/kubectl_kadalu$ python3 __main__.py version
kubectl-kadalu plugin: devel
kadalu pod(s) versions
pod/kadalu-csi-provisioner-0: devel
pod/operator-7dfcc8fb86-ncvzt: devel
```

File: cli/kubectl_kadalu/__main__.py [hunk 2]
```diff
@@ -39,9 +39,54 @@ def get_args():
...
+        utils.command_error(cmd, err.stderr)
+        return None
+
+
 def show_version():
     """Show version information"""
-    print("kubectl-kadalu %s" % VERSION)
+    print("kubectl-kadalu plugin: %s" % VERSION)
+    pods = get_all_kadalu_pods()
+    if pods:
```
Comment (vatsa287) [inline cli/kubectl_kadalu/__main__.py:85]: @aravindavk This check is sufficient right? Or something like k cluster-info is required?

Comment (aravindavk) [reply cli/kubectl_kadalu/__main__.py:85]: Looks good

File: cli/kubectl_kadalu/__main__.py [hunk 3]
```diff
@@ -39,9 +39,54 @@ def get_args():
...
+
+def get_kadalu_version_in_pod(pod):
+    """ Get kadalu version from pods inside kadalu-namespace by exec """
+
+    try:
+
+        cmd = ["kubectl", "exec", "-nkadalu", pod, "--", "bash",
+                "-c", "echo $KADALU_VERSION"]
+
+        if "nodeplugin" in pod or "provisioner" in pod:
```
Comment (aravindavk) [inline cli/kubectl_kadalu/__main__.py:64]: <strike>Keep the pod name as is. When you have more node plugins with different versions then it is not possible to identify which nodeplugin is not upgraded.</strike>

Sorry. Ignore my comment. My bad, didn't realize this is for container name.

**Review by aravindavk — APPROVED** [review]

**Review by leelavg — APPROVED** [review]
LGTM

**Pull request merged**